packages/web-components/@groupby/elements-base/test/unit/base.test.ts

```diff
@@ -1,5 +1,86 @@
-import { expect, stub } from '../utils';
+import { expect, stub, spy } from '../utils';
 import { DummyComponent } from './dummy-component';
+import { dataInitializer } from '../../src';
+
+describe('dataInitializer decorator', () => {
+  const testPropertyName = 'results';
+  let testObj;
+
+  beforeEach(() => {
+    testObj = {
+      init: false,
+      get results() {
+        return this._results;
+      },
+      set results(val) {
+        this._results = val;
+      },
+      _results: ['a', 'b'],
+    };
+  });
+
+  it('should extend a property setter', () => {
+    const originalSet = spy(testObj, 'results', ['set']);
+    dataInitializer('init')(testObj, testPropertyName);
+
+    testObj.results = ['c', 'd'];
+
+    expect(originalSet.set).to.be.called;
+    expect(testObj.results).to.deep.equal(['c', 'd']);
+  });
+
+  it('should not set the initialize property to true on the first set call', () => {
+    dataInitializer('init')(testObj, testPropertyName);
+
+    testObj.results = ['c', 'd'];
+
+    expect(testObj.init).to.be.false;
+  });
+
+  it('should not set the initialize property to true on the first set call', () => {
+    dataInitializer('init')(testObj, testPropertyName);
+
+    testObj.results = ['c', 'd'];
+    testObj.results = ['e', 'f'];
+
+    expect(testObj.init).to.be.true;
+  });
+
+  it('should store the initialization states between multiple instances of the component', () => {
+    const testObj2 = { ...testObj };
+    Object.defineProperty(
+      testObj2,
+      testPropertyName,
+      Object.getOwnPropertyDescriptor(testObj, 'results')
+    );
+    dataInitializer('init')(testObj, testPropertyName);
```

Comment:
`'init'` can go into a variable and be reused in the four instances where it's used.